webpack/assets/javascripts/react_app/common/I18n.js

```diff
@@ -56,16 +66,39 @@ const getLocaleData = () => {
     );
     return { domain: 'app', locale_data: { app: { '': {} } } };
   }
-
-  return locales[locale];
+  return unwrapLocaleDomains(locales, locale);
 };
 
 export const jed = forceSingleton('Jed', () => new Jed(getLocaleData()));
 
+export const strictJed = forceSingleton('strictJed', () => {
+  const options = {
+    ...jed.options,
+    missing_key_callback: (_key, _domain) => {
+      throw new Error('Key not found');
+    },
+  };
+  return new Jed(options);
+});
+
+const multidomain = (strict, fallback) => (...args) => {
+  // eslint-disable-next-line no-unused-vars
+  for (const domain of Object.keys(strictJed.options.locale_data)) {
+    try {
```

Comment:
Js looks fine, is the only way to do this conditional return is with try-catch?

Reply:
I haven't found any other way